cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/BasicPluginInfo/index.tsx

```diff
@@ -95,6 +100,11 @@ const BasicPluginInfoView: React.FC<ICreateContext & WithStyles<typeof styles>>
         setPluginState={setPluginState}
         JSONStatus={JSONStatus}
         setJSONStatus={setJSONStatus}
+        filters={filters}
```

Comment:
We should probably think about putting all these in a single object and passing the object. As a component I don't know if it makes sense to have this much props. The component says JsonMenu .. The first thing I would expect when using the component would be a json which is kind of hidden under all these props..

Reply:
If we move it to an object then we define a type for the object and then ts can validate changes that can happen tomorrow if the contract (props to the component) changes.

Reply:
After future PR #12310 and PR #12291  where we refactor and optimize performance for the Plugin JSON Creator, this issue has been resolved. 

We no longer pass a lot of states to the `JsonMenu`, and the component is only initialized once so that we don't have to pass these many props from each page.
https://github.com/cdapio/cdap/blob/882eb810d3291ed6b39efaf6394a932584abb867/cdap-ui/app/cdap/components/PluginJSONCreator/Create/index.tsx#L271